### ISSUE TITLE
Client-side validation can silently fail on name collision

### ISSUE THREAD

**Issue by aaronadamsCA** [edmundhung/conform#384]
### Describe the bug and the expected behavior

In certain cases, form validation can fail with no apparent error message.

In this contrived example, I've added a read-only input with the same `name` as a text input:

```tsx
<input {...conform.input(foo)} />
<input type="text" name="foo" value="bar" readOnly />
```

When the form is parsed against this schema:

```tsx
const schema = z.object({
  foo: z.string(),
});
```

Validation fails, but no error is reported.

If you switch the order of the inputs, you will see the expected error message `Expected string, received array`.

### Conform version

v0.9.1

### Steps to Reproduce the Bug or Issue

[StackBlitz reproduction](https://stackblitz.com/edit/remix-run-remix-fewxwa?file=app%2Froutes%2F_index.tsx)

Note the two routes, the only difference between them is the order of the fields.

### Additional context

I ran into this organically and couldn't figure out what was happening. It turned out multiple values were colliding, causing client-side validation to fail; but no error was being reported at either the form or field levels. I managed to narrow the issue all the way down to this minimal reproduction.

I have no idea why the order of the fields matters, because the resulting `FormData` seems like it should be identical.

**Comment by jmaldon1** [comment]
I ran into this and wasn't sure if it was zod or conform.

**Comment by edmundhung** [comment]
This is because of the use of `setCustomValidity()` API in the current version and will be resolved with the [new archiecture](https://github.com/edmundhung/conform/discussions/331).

**Comment by aaronadamsCA** [comment]
Confirming this is resolved in `v1.0.0-rc.0`:

https://stackblitz.com/edit/remix-run-remix-auu2yj?file=app%2Froutes%2Fexpected.tsx

Thank you!

**Comment by edmundhung** [comment]
This should be resolved in [v1.0.0](https://github.com/edmundhung/conform/releases/tag/v1.0.0).
